src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java

```diff
@@ -234,17 +234,21 @@ private Map<Long, Long> importTasks(final ProgrammingExercise templateExercise,
      * @param targetExercise   for which static code analysis categories will be copied
      */
     private void importStaticCodeAnalysisCategories(final ProgrammingExercise templateExercise, final ProgrammingExercise targetExercise) {
-        targetExercise.setStaticCodeAnalysisCategories(templateExercise.getStaticCodeAnalysisCategories().stream().map(originalCategory -> {
+        if (targetExercise.getStaticCodeAnalysisCategories() == null) {
+            targetExercise.setStaticCodeAnalysisCategories(new HashSet<>());
+        }
+
+        templateExercise.getStaticCodeAnalysisCategories().forEach(originalCategory -> {
             var categoryCopy = new StaticCodeAnalysisCategory();
             categoryCopy.setName(originalCategory.getName());
             categoryCopy.setPenalty(originalCategory.getPenalty());
             categoryCopy.setMaxPenalty(originalCategory.getMaxPenalty());
             categoryCopy.setState(originalCategory.getState());
             categoryCopy.setProgrammingExercise(targetExercise);
 
-            staticCodeAnalysisCategoryRepository.save(categoryCopy);
-            return categoryCopy;
-        }).collect(Collectors.toSet()));
+            categoryCopy = staticCodeAnalysisCategoryRepository.save(categoryCopy);
+            targetExercise.getStaticCodeAnalysisCategories().add(categoryCopy);
```

Comment:
I'd suggest to create a method `addCodeAnalysisCategories()` to exercise here.

Reply:
Done in 8393aaaab3191224a5e7fb190df924d48a779f6d.